doc/compiler/il/SymbolsSymrefsAliasing.md

```diff
@@ -0,0 +1,171 @@
+bit must be set in all the bit vectors associated with queries that match the new symref.
+
+These bit vectors are only present for primitive queries. Bit vectors for complex queries
+are constructed by `TR:AliasBuilder` by "ORing" together the various bit vectors
+associated with queries that compose the larger query.
+
+### TR_AliasSetInterface
+
+The `TR_AliasSetInterface` class is a wrapper around a bit vector representing aliased
+symrefs. It provides various methods for performing common operations on these bit vectors.
+[The CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) is used
+to achieve static polymorphism.
+
+One key interface method that *must* be overridden by sub-classes is `getTRAliases()`.
+This is the method responsible for returning the bit vector of aliases.
+
+It is worth noting that some of these operations are templated. So, theoretically, they
+can be used on bit vector types different from `TR_BitVector`. However, the type is
+required to satisfy the (unspecified) interface of `TR_BitVector`.
+
+Another noteworthy point is that some of the operations rely on calls to functions in
+the derived class, as is common with the CRTP. However, because of C++ lazy template
+instantiation, these operations will only be materialized if they are called. Hence,
+derived classes are not strictly required to define these called functions, as long
+as the operations that require them are never called on an instance of this class.
+
+There two classes that inherit from this base class, namely `TR_SymAliasSetInterface`
+and `TR_NodeAliasSetInterface`.
+
+### TR_SymAliasSetInterface
+
+`TR_SymAliasSetInterface` acts as a wrapper around a symref and provides an API for
+dealing with aliases of the symref. As should be expected, its `getTRAliases()`
+method returns the set of aliases of the symref.
+
+Being a class template, it has two specializations with different implementations
+of `getTRAliases()`. The non-type template parameter is one of the two values of
+the `TR_AliasSetType` enum: `useDefAliasSet` or `UseOnlyAliasSet`. 
+
+### TR_NodeAliasSetInterface
+
+The `TR_NodeAliasSetInterface` is a wrapper around a node (instance of `TR::Node`).
+It provides an API for dealing with aliases of the symref *contained* by the node.
+That is, its `getTRAliases()` method returns the set of aliases of the symref
+contained by the node.
+
+It too is a template class with two specializations having different implementations
+of `getTRAliases()`. In this case, the non-type template parameter is one of the two
+values of the `TR_NodeAliasSetType` enum: `mayUseAliasSet` or `mayKillAliasSet`.
+
+It should be noted that this class mostly exists for historic reasons. Once upon a time,
+aliasing could apply to instances of `TR::Node` as well as symrefs. However, since this
+is no longer the case, the class is now only used for accessing aliases of the contained
+symref.
+
```

Comment:
We could mention that it is currently used as a shortcut to retrieve the node's symref's alias bit vector?